aten/src/ATen/cuda/CUDAEvent.h

```diff
@@ -79,9 +107,7 @@ struct AT_CUDA_API CUDAEvent {
     if (is_created_) {
       AT_ASSERT(device_index_ == stream.device_index());
```

Comment:
I think this should be an AT_CHECK with a useful error message instead of AT_ASSERT